app/src/main/java/com/eveningoutpost/dexdrip/services/Ob1G5CollectionService.java

```diff
@@ -2100,6 +2101,14 @@ public static List<StatusItem> megaStatus() {
         }
         if ((lastSensorStatus != null)) {
             l.add(new StatusItem("Sensor Status", lastSensorStatus, lastSensorState != Ok ? NOTICE : NORMAL));
+            if (FirmwareCapability.isTransmitterRawIncapable(getTransmitterID())) { // G6 Firefly or G7
+                if (Sensor.isActive()) { // Only if there is an active sensor session in xDrip.
+                    if (Sensor.currentSensor().started_at != DexSessionKeeper.getStart()) { // If xDrip start time differs from transmitter start time
+                        Sensor.updateSensorStartTime(DexSessionKeeper.getStart()); // Change xDrip sensor start time to match that reported by transmitter
+                        UserError.Log.e(TAG, "Updating sensor start time from the transmitter");
+                    }
+                }
+            }
```

Comment:
I don't think megastatus is the right place to update this as that is only triggered when the UI is displayed. I think changes are needed like:

- Don't change start time the difference is already within say 10 minutes of the existing value.
- Don't change the sensor start time if the start time would be more than say a month in the past
- I would functionalize the check and move the call to it to just after the call to `DexTimeKeeper.updateAge()` in ob1 state machine EGlucoseRxMessage2 and also in ClassifierAction in the G7 section.